onvm/onvm_nflib/onvm_common.h

```diff
@@ -495,4 +496,30 @@ whether_wakeup_client(struct onvm_nf *nf, struct nf_wakeup_info *nf_wakeup_info)
 
 #define RTE_LOGTYPE_APP RTE_LOGTYPE_USER1
 
+/*
+ * Updates the ether_addr struct with a fake, safe MAC address.
+ */
+static inline int
+onvm_get_fake_macaddr(struct ether_addr *mac_addr) {
+        uint16_t *mac_addr_bytes = (uint16_t *)((struct ether_addr *)(mac_addr)->addr_bytes);
+        mac_addr_bytes[0] = 2;
+        mac_addr_bytes[1] = 0;
+        mac_addr_bytes[2] = 0;
+        return 0;
+}
+
+/*
+ * Tries to fetch the MAC address of the port_id.
+ * Return 0 if port is valid, -1 if port is invalid.
+ */
+static inline int
+onvm_get_macaddr(uint8_t port_id, struct ether_addr *mac_addr) {
+        if (!rte_eth_dev_is_valid_port(port_id)) {
+                return -1;
+        } else {
+                rte_eth_macaddr_get(port_id, mac_addr);
```

Comment:
Lets check the return value of eth_macaddr_get too just to be safe.

Reply:
The return type of `rte_eth_macaddr_get` is void.
Further, looking at the code [here](https://github.com/DPDK/dpdk/blob/0da7f445df445630c794897347ee360d6fe6348b/lib/librte_ethdev/rte_ethdev.c#L2566-L2574) and [here](https://github.com/DPDK/dpdk/blob/0da7f445df445630c794897347ee360d6fe6348b/lib/librte_ethdev/rte_ethdev.h#L1232-L1237), they too use `rte_eth_dev_is_valid_port` to throw errors and thus, this should be fine.

Reply:
sounds good, my bad for not catching that